pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableInstances.java

```diff
@@ -159,9 +161,11 @@ public List<String> getLiveBrokersForTable(
   @ApiResponses(value = {
       @ApiResponse(code = 200, message = "Success"), @ApiResponse(code = 500, message = "Internal server error")
   })
-  public Map<String, List<InstanceInfo>> getLiveBrokers() {
+  public Map<String, List<InstanceInfo>> getLiveBrokers(
+          @ApiParam(value = "Table name (with or without type)", required = false) @DefaultValue("")
+          @QueryParam("tableName") String tableName) {
     try {
-      return _pinotHelixResourceManager.getTableToLiveBrokersMapping();
+      return _pinotHelixResourceManager.getTableToLiveBrokersMapping(Optional.of(tableName));
```

Comment:
This will not give `Optional.empty()` when tableName is empty which might be the desired behaviour here. You can modify it to - `tableName.equals("") ? Optional.empty() : Optional.of(tableName)` ?

Reply:
The check `if (optionalTableName.isPresent() && !optionalTableName.get().isEmpty())` in `getTableToLiveBrokersMapping` checks if the tableName string is empty. 
Added a test case for the same.